File: ruby/hamming/hamming.rb
module Hamming
  class << self
    def compute(first, second)
      raise ArgumentError if first.length() != second.length()

      first.chars.zip(second.chars).reduce(0) do |diff, pair|
        pair.first == pair.last ? diff : diff + 1
      end
    end
  end
end

module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(first, second) {
  if (first.length !== second.length) {
    throw new Error('DNA strands must be of equal length.');
  }

  var diff = 0;

  for (var i = 0; i < first.length; i++) {
    if (first.charAt(i) !== second.charAt(i)) {
      diff += 1;
    }
  }

  return diff;
};

module.exports = Hamming;
